src/common.c

```diff
@@ -127,47 +127,32 @@ raft_node_t *getLeaderNodeOrReply(RedisRaftCtx *rr, RedisModuleCtx *ctx)
     return node;
 }
 
-/* Checks that we *have* a Raft leader and its identity.
- *
- * If no leader is currently elected, replies a -CLUSTERDOWN error and
- * return RR_ERROR.
- *
- * If ret_leader is provided, it is used to return the current leader
- * by reference along with an RR_OK return value.
+/* Returns the leader node (Node), or reply a -CLUSTERDOWN error
+ * and return NULL.
  *
- * If ret_leader is not provided and the current node is NOT the leader,
- * a -MOVED reply is generated with the current leader and RR_ERROR is
- * returned.
+ * Note: it's possible to have an elected but unknown leader node, in which
+ * case NULL will also be returned.
  */
-RRStatus checkLeader(RedisRaftCtx *rr, RedisModuleCtx *ctx, Node **ret_leader)
+Node *getLeaderNodeOrReply(RedisRaftCtx *rr, RedisModuleCtx *ctx)
 {
-    Node *node = NULL;
-
-    if (raft_is_leader(rr->raft)) {
-        return RR_OK;
-    }
-
-    raft_node_t *leader = raft_get_leader_node(rr->raft);
-    if (leader) {
-        node = raft_node_get_udata(leader);
+    raft_node_t *raft_node = getLeaderRaftNodeOrReply(rr, ctx);
+    if (!raft_node) {
+        return NULL;
     }
 
+    Node *node = raft_node_get_udata(raft_node);
     if (!node) {
         RedisModule_ReplyWithError(ctx, err_clusterdown);
-        return RR_ERROR;
     }
 
-    if (ret_leader) {
-        *ret_leader = node;
-        return RR_OK;
-    }
+    return node;
+}
 
-    /* One anomaly here is that may redirect a client to the leader even for
-     * commands have no keys, which is something Redis Cluster never does. We
-     * still need to consider how this impacts clients which may not expect it.
-     */
-    replyRedirect(ctx, 0, &node->addr);
-    return RR_ERROR;
+
+/* Checks if the current cluster is the leader */
+bool isLeader(RedisRaftCtx *rr) {
```

Comment:
Sorry for nitpick, as this function doesn't do anything on top of `raft_is_leader()`, probably we don't need it.